pkg/system/reconciler.go

```diff
@@ -205,6 +209,12 @@ func NewReconciler(
 	r.SecretServer.StringData["jwt"] = util.RandomBase64(16)
 	r.SecretServer.StringData["server_secret"] = util.RandomHex(4)
 
+	// set noobaa root master key secret
+	r.NooBaaRootSecret.Data["cipher_iv"] = util.RandomBytes(16)
+	r.NooBaaRootSecret.Data["cipher_key"] = util.RandomBytes(32)
```

Comment:
So I understand that you are using .Data instead of .StringData to store the raw random binary data, and avoid double base64 encoding, because kubernetes already stores it the secret encoded in base64.

One concern that I have with this is if it merges the two maps or just replaces Data when we define StringData... 

The other concern I have is that setting raw binary data in the secret may lead to unprintable StringData...

My instinct is to prefer to use double base64 in this case - and even make the key names represent it clearly that we are communicating b64 encoded data.

```suggestion
	r.NooBaaRootSecret.StringData["cipher_iv_b64"] = util.RandomBase64(16)
	r.NooBaaRootSecret.StringData["cipher_key_b64"] = util.RandomBase64(32)
```

Reply:
np, fixed.
also, moved other fields beside cipher_key to the core part (IV removed completely - not needed for the first encryption of the system master key) and updated the endpoint deployment and statefulset-core to set the cipher_key_b64 to env variable.